### PR TITLE
Add a depth parameter to Camera::project_position()

### PR DESCRIPTION
**Pull request by Cheeseness**
Camera::project_position()'s current implementation makes assumptions that seem to work against the practical value of a function for converting 2D viewport coordinates into 3D coordinates. The current undocumented behaviour of returning a point on the camera's near clipping plane doesn't align with any use-cases that I can come up with, and is far enough from intuitive that I had to read the code to work out what was going on.

Having the depth exposed as a parameter allows the near clipping plane to be passed in if that ever happens to be desirable, while making the function more broadly usable without having to [temporarily change the near clipping plane position](https://twitter.com/ValiantCheese/status/1131420463060533248).

In case preserving the current behaviour is desirable, I'd be happy to look at that, but I'm not sure how to approach function overloading with the language binding implementation.

### CODE REVIEW

**Comment by reduz** [comment]
This is a strange function that was there since the beginning that served no real purpose (the actual one that is used everywhere is project_ray_position). It also does not match the sister function unproject_position, as it adds a near value that is unwanted.

I think the PR is good, but depth should be 0 by default

**Comment by Cheeseness** [comment]
> I think the PR is good, but depth should be 0 by default

Thanks. Updated to default to 0.

File: doc/classes/Camera.xml [hunk 1]
```diff
@@ -75,9 +75,10 @@
 			<return type="Vector3">
 			</return>
 			<argument index="0" name="screen_point" type="Vector2">
+			<argument index="1" name="z_depth" type="float" default="0">
```
Comment (akien-mga) [inline doc/classes/Camera.xml]: There should be a `</argument>` before this line. The docs XML can be generated/validated with `godot --doctool .` from your build folder.

File: scene/3d/arvr_nodes.cpp [hunk 2]
```diff
@@ -127,15 +127,15 @@ Point2 ARVRCamera::unproject_position(const Vector3 &p_pos) const {
 	return res;
 };
 
-Vector3 ARVRCamera::project_position(const Point2 &p_point) const {
+Vector3 ARVRCamera::project_position(const Point2 &p_point, float z_depth) const {
```
Comment (akien-mga) [inline scene/3d/arvr_nodes.cpp]: Parameters should start with `p_`, so even though it's a bit awkward it should be `p_z_depth` here.

**Comment by owstetra** [comment]
@reduz we use this function in our game , we use it to spawn a Waypoint on the Minimap

File: scene/3d/arvr_nodes.cpp (hunk 2 above)
Comment (Cheeseness) [reply scene/3d/arvr_nodes.cpp]: Ah, thank you. That has been corrected.

File: doc/classes/Camera.xml (hunk 1 above)
Comment (Cheeseness) [reply doc/classes/Camera.xml]: My bad! Should be sorted now.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks for your contribution!

**Comment by Cheeseness** [comment]
Glad to pitch in. Thanks for merging :)

**Pull request merged**